brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -97,21 +96,73 @@
  * }</pre>
  */
 public final class ExtraFieldPropagation<K> implements Propagation<K> {
+  static final FieldCustomizer NOOP_FIELD_CUSTOMIZER = new FieldCustomizer() {
+    @Override public String update(String key, String value) {
+      return value;
+    }
+  };
+  static final Customizer[] NO_CONTEXT_CUSTOMIZERS = new Customizer[0];
+  static final FieldCustomizer[] NO_FIELD_CUSTOMIZERS = new FieldCustomizer[0];
+
+  public interface FieldCustomizer {
+    /**
+     * Returns the same value, an updated one, or null to drop the field.
+     *
+     * @see ExtraFieldPropagation#set(String, String)
+     */
+    @Nullable String update(String fieldName, @Nullable String value);
+  }
+
+  /**
+   * Methods here are called once per extract or inject, which allows the returned customizer to
+   * handle multiple fields as needed.
+   */
+  public static abstract class Customizer {
+    public FieldCustomizer extractCustomizer(TraceContextOrSamplingFlags.Builder builder) {
+      return NOOP_FIELD_CUSTOMIZER;
+    }
+
+    public FieldCustomizer injectCustomizer(TraceContext context) {
+      return NOOP_FIELD_CUSTOMIZER;
+    }
+  }
+
+  static final class RedactOnInject extends Customizer implements FieldCustomizer {
```

Comment:
@marcingrzejszczak fyi there's a new customizer coming to town. This can implement things like redaction or truncating extra fields.